packages/db/src/repositories/trade.ts

```diff
@@ -398,6 +432,61 @@ export class TradeRepository {
     }
   }
 
+  /**
+   * Get the latest on-chain trade block number for an agent in a competition on a specific chain
+   * Used to determine incremental sync starting point for spot live competitions
+   * @param agentId Agent ID
+   * @param competitionId Competition ID
+   * @param specificChain Specific chain to query
+   * @returns Latest block number or null if no trades exist
+   */
```

Comment:
Docs repeated here for some reason?

Reply:
also handled in https://github.com/recallnet/js-recall/pull/1596 at [885415e](https://github.com/recallnet/js-recall/pull/1596/commits/885415e4cccb949502f35a294f98d9204369d858)